phoenix-core/src/main/java/org/apache/phoenix/schema/transform/TransformMaintainer.java

```diff
@@ -0,0 +1,459 @@
+
+        this.newTableColumnCount = oldTablePKColumns.size();
+
+        this.newTableColumnTypes = Lists.<PDataType>newArrayListWithExpectedSize(nNewTablePKColumns);
+        this.newTableExpressions = Lists.newArrayListWithExpectedSize(nNewTableColumns);
+        this.coveredColumnsMap = Maps.newHashMapWithExpectedSize(nNewTableColumns - nNewTablePKColumns);
+        this.nNewTableSaltBuckets = newTable.getBucketNum() == null ? 0 : newTable.getBucketNum();
+        this.oldTableEmptyKeyValueCF = SchemaUtil.getEmptyColumnFamily(oldTable);
+        this.emptyKeyValueCFPtr = SchemaUtil.getEmptyColumnFamilyPtr(newTable);
+        this.nOldTableCFs = oldTable.getColumnFamilies().size();
+        this.newTableWALDisabled = newTableWALDisabled;
+        this.newTableImmutableRows = newTable.isImmutableRows();
+        this.newTableColumnsInfo = Sets.newHashSetWithExpectedSize(nNewTableColumns - nNewTablePKColumns);
+
+        for (int i = 0; i < newTable.getColumnFamilies().size(); i++) {
+            PColumnFamily family = newTable.getColumnFamilies().get(i);
+            for (PColumn newColumn : family.getColumns()) {
+                PColumn oldColumn = getColumnOrNull(oldTable, newColumn.getName().getString(), newColumn.getFamilyName().getString());
+                // This can happen during deletion where we don't need covered columns
+                if (oldColumn != null) {
+                    byte[] oldColumnCq = oldColumn.getColumnQualifierBytes();
+                    byte[] newColumnCq = newColumn.getColumnQualifierBytes();
+                    this.coveredColumnsMap.put(new ColumnReference(oldColumn.getFamilyName().getBytes(), oldColumnCq),
+                            new ColumnReference(newColumn.getFamilyName().getBytes(), newColumnCq));
+                }
+            }
+        }
+        this.logicalNewTableName = newTable.getName().getString();
+        initCachedState();
+    }
+
+    public static PColumn getColumnOrNull(PTable table, String columnName, String familyName) {
+        PColumnFamily family;
+        try {
+            family = table.getColumnFamily(familyName);
+        } catch (ColumnFamilyNotFoundException e) {
+            return null;
+        }
+        try {
+            return family.getPColumnForColumnName(columnName);
+        } catch (ColumnNotFoundException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Init calculated state reading/creating
+     */
+    private void initCachedState() {
+        byte[] newTableEmptyKvQualifier = EncodedColumnsUtil.getEmptyKeyValueInfo(newTableEncodingScheme).getFirst();
+        byte[] oldTableEmptyKvQualifier = EncodedColumnsUtil.getEmptyKeyValueInfo(oldTableEncodingScheme).getFirst();
+        newTableEmptyKeyValueRef = new ColumnReference(oldTableEmptyKeyValueCF, newTableEmptyKvQualifier);
+        oldTableEmptyKeyValueRef = new ColumnReference(oldTableEmptyKeyValueCF, oldTableEmptyKvQualifier);
```

Comment:
Planning to use cached value of `oldTableEmptyKeyValueRef` in future?

Reply:
Yes. We might need it with the new transform types. If you prefer, I can remove now and add when we need it